exdb/views.py

```diff
@@ -108,18 +106,22 @@ def form_valid(self, form):
         form.instance.experience = self.get_experience()
         form.instance.experience.status = 'ad' if self.request.POST.get('approve') else 'de'
         if form.instance.experience.status == 'ad':
-            form.instance.experience.approver = self.request.user
-            form.instance.experience.approved_timestamp = now()
+            form.instance.experience.current_approver = self.request.user
+            ExperienceApproval.objects.create(experience=form.instance.experience,
+                                              approver=self.request.user,
+                                              timestamp=now())
         form.instance.experience.save()
         return super(ExperienceApprovalView, self).form_valid(form)
 
     def form_invalid(self, form):
         if self.request.POST.get('approve'):
             experience = self.get_experience()
-            experience.approver = self.request.user
-            experience.approved_timestamp = now()
+            experience.current_approver = self.request.user
             experience.status = 'ad'
             experience.save()
+            ExperienceApproval.objects.create(experience=experience,
+                                              approver=self.request.user,
+                                              timestamp=now())
```

Comment:
Setting the `timestamp` here is unnecessary since it will get set to `now()` automatically.